crates/kubelet/src/module_store.rs

```diff
@@ -12,6 +12,40 @@ use std::convert::TryFrom;
 use std::path::{Path, PathBuf};
 use std::sync::Arc;
 
+/// Specifies how the store should check for module updates
+#[derive(PartialEq, Debug)]
+pub enum ModulePullPolicy {
+    /// Always return the module as it currently appears in the
+    /// upstream registry
+    Always,
+    /// Return the module as it is currently cached in the local store if
+    /// present; fetch it from the upstream registry only if it it not
+    /// present in the local store
+    IfNotPresent,
+    /// Never fetch the module from the upstream registry; if it is not
+    /// available locally then return an error
+    Never,
+}
+
+impl ModulePullPolicy {
+    /// Parses a module pull policy from a Kubernetes ImagePullPolicy string
+    pub fn parse(name: Option<String>) -> anyhow::Result<Option<ModulePullPolicy>> {
+        match name {
+            None => Ok(None),
+            Some(n) => ModulePullPolicy::parse_str(&n[..]),
+        }
+    }
+
+    fn parse_str(name: &str) -> anyhow::Result<Option<ModulePullPolicy>> {
+        match name {
```

Comment:
Should this be case sensitive?
i.e. - if I try to set the pull policy as "always", I get an error:

```
---- module_store::test::can_parse_pull_policies stdout ----
thread 'module_store::test::can_parse_pull_policies' panicked at 'called `Result::unwrap()` on an `Err` value: unrecognized pull policy always', crates/kubelet/src/note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
I think kubernetes expects capitalized

Reply:
Just checked the Kubelet code, you are right - https://github.com/kubernetes/kubernetes/blob/9cd906e9322f75ce48b47613dd9e025c2b06c143/pkg/kubelet/images/image_manager.go#L65-L76